language/tools/move-mv-llvm-compiler/src/stackless/module_context.rs

```diff
@@ -772,6 +799,21 @@ impl<'mm, 'up> ModuleContext<'mm, 'up> {
                     let attrs = self.mk_pattrs_for_move_type(1);
                     (llty, attrs)
                 }
+                "str_cmp_eq" => {
+                    // str_cmp_eq(str1_ptr: &AnyValue, str1_len: &AnyValue,
+                    //            str2_ptr: &AnyValue, str1_len: &AnyValue) -> bool
+                    let ret_ty = llcx.int_type(1);
+                    let ptr_ty = llcx.int_type(8).ptr_type();
+                    let len_ty = llcx.int_type(64);
+                    let param_tys = &[ptr_ty, len_ty, ptr_ty, len_ty];
+                    let llty = llvm::FunctionType::new(ret_ty, param_tys);
+                    let mut attrs = self.mk_pattrs_for_move_type(1);
```

Comment:
Either use `mk_pattrs_for_move_type` or below, but not both. The former was meant to add attrs for `MoveType`, but in this case, it will work also. That is, you created what amounts to a descriptor with static/constant values. So other than the name not being entirely accurate for this case, `mk_pattrs_for_move_type` is more concise.

Reply:
Fixed. Thanks for catching this!